content/en/smart-contracts/fundamentals/programmatic-storage/index.md

```diff
@@ -0,0 +1,60 @@
+    identifier: "programmatic-storage"
+weight: 110
+toc: true
+---
+
+## Manual storage
+
+A client must go through several steps to manually store something with a storage provider on Filecoin without using the FVM. These steps involve interacting with the Filecoin network, negotiating a deal with a storage provider, and performing various other operations to ensure the data is stored correctly and securely. Let's break down the process in detail:
+
+1. **Identify a storage provider (SP)**: Clients must identify storage providers who will store their data. Clients usually pick their storage providers based on geographic location and bandwidth limitations, although other factors may impact their decision.
+1. **Negotiate a deal**: Clients negotiate deals with storage providers. These deals include terms like data size, price, deal duration, and collateral. The deal-making process happens off-chain, and once both parties agree to the terms of a deal, that deal is published on-chain for the rest of the network to see and validate. To successfully store data on Filecoin, clients need to have enough FIL to cover the deal's cost.
+1. **Data preparation**: Before storing files on Filecoin, clients must import them into their local Filecoin node, which produces a data CID (Content Identifier) - an ID uniquely describing the content. Then, the data is transferred to the storage provider.
+1. **Deal publication**: The deal is published on-chain, making the storage provider publicly accountable for the deal.
+1. Sealing and proving storage**: The storage provider stores the data in sectors. Sectors are the basic units of provable storage where storage providers store clients’ data and generate Proof-of-Spacetime (PoSt) on behalf of the Filecoin network. Sectors have standard sizes and a lifetime that storage providers can extend before reaching the end of the lifetime.
+
+## Programmatic storage
+
+The Filecoin virtual machine (FVM) is not designed to directly interact with data on Filecoin. Instead, it operates on blockchain state data, which includes metadata about stored data. The FVM's role is to compute over this state data and create logic around the storage and retrieval process. Storage and retrieval providers on the Filecoin network handle the actual process of storing and retrieving data.
+
+Develoeprs can use FVM actors (also called smart-contracts) to implement various use cases related to data storage and retrieval on the Filecoin network. For instance, they can automate the process of renewing storage deals, replicate data across multiple storage providers for redundancy, or implement access control rules for data retrieval.
+
+The FVM also includes an Ethereum Virtual Machine (EVM) runtime, enabling developers to deploy existing EVM-based smart contracts onto the FVM with little to no modifications. If you already have a smart contract written in Solidity, you can deploy it to Filecoin and have it interact with the Filecoin storage system.
```

Comment:
can we reduce this by saying "FVM is fully Ethereum-compatible [hyperlink to https://docs.filecoin.io/smart-contracts/fundamentals/filecoin-evm-runtime/] and today, smart contracts on Solidity can be deployed to Filecoin."

Reply:
On second review, this sentence doesn't add much to this page. The FVM-EVM relationship is discussed, in length, on other pages. I'll remove this paragraph.